catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -331,31 +319,41 @@ public UpdateResponse update(UpdateRequest updateRequest) throws IngestException
             .map(Serializable::toString)
             .collect(Collectors.toList());
 
-    /* 1a. Create the old Metacard Query */
-    String attributeQuery = getQuery(attributeName, identifiers);
-
-    SolrQuery query = new SolrQuery(attributeQuery);
-    // Set number of rows to the result size + 1.  The default row size in Solr is 10, so this
-    // needs to be set in situations where the number of metacards to update is > 10.  Since there
-    // could be more results in the query response than the number of metacards in the update
-    // request,
-    // 1 is added to the row size, so we can still determine whether we found more metacards than
-    // updated metacards provided
-    query.setRows(updates.size() + 1);
-    QueryResponse idResults = null;
+    final Map<Serializable, Metacard> idToMetacardMap = new HashMap<>();
+    if (Metacard.ID.equals(attributeName)) {
+      try {
+        idToMetacardMap.putAll(
+            client
+                .getIds(new HashSet<>(identifiers))
+                .stream()
+                .filter(Objects::nonNull)
+                .collect(Collectors.toMap(Metacard::getId, Function.identity())));
+      } catch (UnsupportedQueryException e) {
+        LOGGER.info("Failed to query for metacard(s) by ID before update.", e);
```

Comment:
✏️ Exception stack traces should only be logged at debug level otherwise it pollutes the log. Can we add a second debug trace that would trace the stack trace and keep the INFO log with only the toString() from `e`?